Wikipedia/Code/ColumnarCollectionViewController.swift

```diff
@@ -257,6 +257,13 @@ class ColumnarCollectionViewController: ViewController, ColumnarCollectionViewLa
         previewingContext.sourceRect = view.convert(cell.bounds, from: cell)
         return indexPath
     }
+
+    // MARK: - Event logging utiities
+
+    var maxViewed: NSNumber {
+        let value = ((Double(collectionView.contentOffset.y) + Double(collectionView.bounds.height)) / Double(collectionView.contentSize.height)) * 100
```

Comment:
If the user scrolls to the bottom (100%), then scrolls half way back up (50%) and selects an item to view, this will be 50% instead of 100%

Reply:
You should be able to override `scrollViewDidScroll` as long as it still calls super. You might also want to ensure `contentSize.height.isGreaterThan` is greater than 0 and that you aren't returning percentages higher than 100

Reply:
why should it be 100% if they scrolled back up to 50%?

Reply:
nvm I missed `max` in `maxViewed` 😂